src/Tasks/Microsoft.NET.Build.Tasks/Microsoft.NET.Build.Tasks.csproj

```diff
@@ -54,10 +57,8 @@
     <PackageReference Include="NuGet.ProjectModel" />
     <PackageReference Include="NuGet.Build.Tasks.Pack" ExcludeAssets="All" />
     <PackageReference Include="Microsoft.Deployment.DotNet.Releases" />
-    <ProjectReference Include="..\..\ApiCompat\Microsoft.DotNet.ApiCompatibility\Microsoft.DotNet.ApiCompatibility.csproj" OutputItemType="ReferenceCopyLocalPaths" ReferenceOutputAssembly="false" />
-    <ProjectReference Include="..\..\ApiCompat\Microsoft.DotNet.PackageValidation\Microsoft.DotNet.PackageValidation.csproj" OutputItemType="ReferenceCopyLocalPaths" ReferenceOutputAssembly="false" />
-    <ProjectReference Include="..\..\ApiCompat\Microsoft.DotNet.ApiCompat.Task\Microsoft.DotNet.ApiCompat.Task.csproj" OutputItemType="ReferenceCopyLocalPaths" ReferenceOutputAssembly="false" />
-    <ProjectReference Include="..\..\Microsoft.DotNet.ApiSymbolExtensions\Microsoft.DotNet.ApiSymbolExtensions.csproj" OutputItemType="ReferenceCopyLocalPaths" ReferenceOutputAssembly="false" />
+    <!-- Use an alias for APICompat so that we bring in the binaries, but don't accidentally reference any types -->
+    <ProjectReference Include="..\..\ApiCompat\Microsoft.DotNet.ApiCompat.Task\Microsoft.DotNet.ApiCompat.Task.csproj" Aliases="unused" PrivateAssets="All" />
```

Comment:
Interesting. I didn't know about the `Aliases` metadata. Is that documented somewhere? Do other components use a P2P like this / is this well supported by msbuild?

Reply:
Aliases have been around forever.  Support for Aliases metadata is a compiler feature, the only thing it does is to scope the reference passed to the compiler.
https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-options/inputs#references
Aliases is [documented](https://learn.microsoft.com/en-us/visualstudio/msbuild/common-msbuild-project-items?view=vs-2022#reference) metadata on reference.  It works for ProjectReferences in the same way because it will flow to the ultimate ReferencePath items passed to the compiler.

I could remove that and instead just `NoWarn` the conflicting type definition of TaskBase but that felt less clean here.  I like Alias because it makes the build treat it like any other reference - allowing it's dependency closure to participate in restore - but makes it so it doesn't accidentally get referenced by the SDK's task assembly.